org.osgi.test.common/src/main/java/org/osgi/test/common/annotation/config/Configuration.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.annotation.config;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Inherited
+@Target({
+	TYPE, METHOD, PARAMETER, FIELD
+})
+@Repeatable(Configurations.class)
+@Retention(RUNTIME)
+@Documented
+public @interface Configuration {
```

Comment:
If we follow the pattern used in our other injector extensions, we should call this `@InjectConfiguration` rather than simply `@Configuration`. This would also help to avoid the name collision with `org.osgi.service.cm.Configuration`, which I can forsee getting annoying for users of this extension.